internal/cephfs/core/quiesce.go

```diff
@@ -0,0 +1,240 @@
+// ceph fs quiesce expects.
+// Example: ["group1/subvolume1", "group1/subvolume2", "group2/subvolume1"].
+func (fq *fsQuiesce) getMembers() []string {
+	volName := []string{}
+	for svg, sb := range fq.subVolumeGroupMapping {
+		for _, s := range sb {
+			// name is in the format of subvolume/group
+			// TODO: Remove this hardcoding as it is not required when we get next release of ceph
+			name := svg + "/" + s + "?q=30"
+			volName = append(volName, name)
+		}
+	}
+
+	return volName
+}
+
+func (fq *fsQuiesce) FSQuiesce(
+	ctx context.Context,
+	reserveName string,
+) (*admin.QuiesceInfo, error) {
+	opt := &admin.FSQuiesceOptions{
+		Timeout:    180,
+		AwaitFor:   0,
+		Expiration: 180,
+	}
+	log.DebugLog(ctx,
+		"FSQuiesce for reserveName %s: members:%v options:%v",
+		reserveName,
+		fq.getMembers(),
+		opt)
+	resp, err := fq.fsa.FSQuiesce(fq.fsName, admin.NoGroup, fq.getMembers(), reserveName, opt)
+	if resp != nil {
+		qInfo := resp.Sets[reserveName]
+
+		return &qInfo, nil
+	}
+
+	log.ErrorLog(ctx, "failed to quiesce filesystem %s", err)
+
+	return nil, err
+}
+
+func (fq *fsQuiesce) FSQuiesceWithExpireTimeout(ctx context.Context,
+	reserveName string,
+) (*admin.QuiesceInfo, error) {
+	opt := &admin.FSQuiesceOptions{
+		Timeout:    180,
+		AwaitFor:   0,
+		Expiration: 180,
+	}
+	log.DebugLog(ctx,
+		"FSQuiesceWithExpireTimeout for reserveName %s: members:%v options:%v",
+		reserveName,
+		fq.getMembers(),
+		opt)
+	resp, err := fq.fsa.FSQuiesce(fq.fsName, admin.NoGroup, fq.getMembers(), reserveName, opt)
+	if resp != nil {
+		qInfo := resp.Sets[reserveName]
+
+		return &qInfo, nil
+	}
+
+	log.ErrorLog(ctx, "failed to quiesce filesystem with expire timeout %s", err)
+
+	return nil, err
+}
+
+func (fq *fsQuiesce) ResetFSQuiesce(ctx context.Context,
+	reserveName string,
+) (*admin.QuiesceInfo, error) {
+	opt := &admin.FSQuiesceOptions{
+		Reset:      true,
+		AwaitFor:   0,
+		Timeout:    180,
+		Expiration: 180,
+	}
+	// Reset the filesystem quiesce so that the timer will be reset, and we can
+	// reuse the same reservation if it has already failed or timed out.
+	log.DebugLog(ctx,
+		"ResetFSQuiesce for reserveName %s: members:%v options:%v",
+		reserveName,
+		fq.getMembers(),
+		opt)
+	resp, err := fq.fsa.FSQuiesce(fq.fsName, admin.NoGroup, fq.getMembers(), reserveName, opt)
+	if resp != nil {
+		qInfo := resp.Sets[reserveName]
+
+		return &qInfo, nil
+	}
+
+	log.ErrorLog(ctx, "failed to reset timeout for quiesce filesystem %s", err)
+
+	return nil, err
+}
+
+func (fq *fsQuiesce) ReleaseFSQuiesce(ctx context.Context,
+	reserveName string,
+) (*admin.QuiesceInfo, error) {
+	opt := &admin.FSQuiesceOptions{
+		AwaitFor: 0,
+		Release:  true,
+	}
+	log.DebugLog(ctx,
+		"ReleaseFSQuiesce for reserveName %s: members:%v options:%v",
+		reserveName,
+		fq.getMembers(),
+		opt)
+	resp, err := fq.fsa.FSQuiesce(fq.fsName, admin.NoGroup, []string{}, reserveName, opt)
+	log.DebugLog(ctx, "error and response", resp, err)
```

Comment:
```suggestion
	log.DebugLog(ctx, "error and response", resp, err)
```

This line may need to be removed if it was added for debugging during coding or error message may need to be improved.

Reply:
it was added for testing, removed it.